library/clinical_rules.php

```diff
@@ -1126,6 +1350,101 @@ function buildPatientArray($patient_id = '', $provider = '', $pat_prov_rel = 'pr
     }
 }
 
+/**
+ * Process patient array that is to be tested. This uses the patient relationship context of encounters linked to billing facilities
+ *
+ * @param  integer       $start         applicable patient to start at (when batching process)
+ * @param  integer       $batchSize     number of patients to batch (when batching process)
+ * @param  boolean       $onlyCount     If true, then will just return the total number of applicable records (ignores batching parameters)
+ * @param  integer       $billing_facility id of the billing facility to constrain patient relationships to, if NULL it uses ALL billing facilities
+ * @param  integer|null  $provider_id   The id of a provider to restrict patient data to if we have one
+ * @return array/integer                Array of patient pid values or number total pertinent patients (if $onlyCount is TRUE)
+ */
+function buildPatientArrayEncounterBillingFacility($start, $batchSize, $onlyCount, $billing_facility, $provider_id = null)
+{
+    $sql = "SELECT DISTINCT `pid` FROM `form_encounter` ";
+    $binds = [];
+
+    $billing_facility = intval($billing_facility); // make sure its an integer
+    if (empty($billing_facility)) {
+        $sql .= "WHERE `billing_facility` IS NOT NULL ";
+    } else {
+        //            " WHERE (`provider_id`=? OR `supervisor_id`=?) AND `billing_facility` = ? ORDER BY `pid`", array($provider,$provider, $billing_facility));
+        $sql .= " WHERE `billing_facility` = ? ";
+        $binds[] = $billing_facility;
+    }
+    if (!empty($provider_id)) {
+        $sql .= " AND provider_id = ? ";
+        $binds[] = $provider_id;
+    }
+    $sql .= "ORDER BY `pid`";
+    if (!($start == null || $batchSize == null || $onlyCount)) {
+        $sql .= "LIMIT ?,?";
```

Comment:
limits can's use binding. i escape with the escape_limit() function (always available and in formdata.inc.php), but intval is ok also since this is what escape_limit basically does